code/object/objcollide.cpp

```diff
@@ -1038,3 +1040,107 @@ void obj_sort_and_collide(SCP_vector<int>* Collision_list)
 	}
 	obj_find_overlap_colliders(sort_list_y, sort_list_z, 2, true);
 }
+
+bool test_line_of_sight(vec3d* from, vec3d* to, std::unordered_set<const object*>&& excluded_objects, bool test_for_shields, bool test_for_hull, float* first_intersect_dist, float threshold) {
```

Comment:
This function is a higher level function unlike the other functions in this file. It would be better to move this to a more object specific place like the object folder and possibly into its own file.

Reply:
I'd suggest keeping it in aicode.cpp, even if it's now a separate function.